sdk/python/kfp/dsl/pipeline_config.py

```diff
@@ -13,11 +13,35 @@
 # limitations under the License.
 """Pipeline-level config options."""
 
+from typing import Any, Dict, Optional
+
+class KubernetesWorkspaceConfig:
+    """Configuration for Kubernetes-specific workspace settings."""
+
+    def __init__(self, pvc_spec_patch: Optional[Dict[str, Any]] = None):        
+        self.pvc_spec_patch = pvc_spec_patch or {}
+
+    def set_pvc_spec_patch(self, patch: Dict[str, Any]):        
+        self.pvc_spec_patch = patch
+
+class WorkspaceConfig:
+    """Configuration for a shared workspace that persists during the pipeline run."""
+
+    def __init__(self, size: str = "", kubernetes: Optional[KubernetesWorkspaceConfig] = None):
```

Comment:
I think we should make `size` required.
```suggestion
    def __init__(self, size: str, kubernetes: Optional[KubernetesWorkspaceConfig] = None):
```